fhir-server/src/main/java/com/ibm/fhir/server/rest/FHIRRestInteractionVisitorOffload.java

```diff
@@ -0,0 +1,246 @@
+
+            // Try offloading storage of the payload. The offloadResponse will be null if not supported
+            String resourcePayloadKey = UUID.randomUUID().toString();
+            int newVersionNumber = Integer.parseInt(resource.getMeta().getVersionId().getValue());
+            PayloadPersistenceResponse actualOffloadResponse = storePayload(resource, resource.getId(), newVersionNumber, resourcePayloadKey);
+
+            // Resource doesn't change, so no need to return it
+            return new FHIRRestOperationResponse(null, null, actualOffloadResponse);
+        });
+    }
+
+    @Override
+    public FHIRRestOperationResponse doInvoke(String method, int entryIndex, Entry validationResponseEntry, String requestDescription, FHIRUrlParser requestURL, long accumulatedTime, FHIROperationContext operationContext, String resourceTypeName, String logicalId,
+            String versionId, Resource resource, MultivaluedMap<String, String> queryParameters) throws Exception {
+        // NOP
+        return null;
+    }
+
+    @Override
+    public FHIRRestOperationResponse doDelete(int entryIndex, String requestDescription, FHIRUrlParser requestURL, long accumulatedTime, String type, String id, String searchQueryString) throws Exception {
+        // NOP
+        return null;
+    }
+
+    @Override
+    public FHIRRestOperationResponse validationResponse(int entryIndex, Entry validationResponseEntry, String requestDescription, long accumulatedTime) throws Exception {
+        // NOP
+        return null;
+    }
+
+    @Override
+    public FHIRRestOperationResponse issue(int entryIndex, String requestDescription, long accumulatedTime, Status status, Entry responseEntry) throws Exception {
+        // NOP
+        return null;
+    }
+
+    /**
+     * If payload offloading is supported by the persistence layer, store the given resource. This
+     * can be an async operation which we resolve at the end just prior to the transaction being
+     * committed. If offloading isn't enabled, the operation is a NOP and the persistence layer 
+     * returns null. 
+     * @param resource
+     * @param logicalId
+     * @param newVersionNumber
+     * @param resourcePayloadKey
+     * @return
+     */
+    protected PayloadPersistenceResponse storePayload(Resource resource, String logicalId, int newVersionNumber, String resourcePayloadKey) throws Exception {
+       return helpers.storePayload(resource, logicalId, newVersionNumber, resourcePayloadKey);
+    }
+
+    /**
+     * Unified exception handling for each of the operation calls
+     * @param entryIndex
+     * @param v
+     * @param failFast
+     * @param requestDescription
+     * @param accumulatedTime
+     * @throws Exception
+     */
+    private FHIRRestOperationResponse doOperation(int entryIndex, String requestDescription, long accumulatedTime, Callable<FHIRRestOperationResponse> v) throws Exception {
```

Comment:
maybe `doInteraction` instead?

Reply:
Would need to also change this in all the other visitors, plus that means renaming FHIRRestOperationResponse to FHIRRestInteractionResponse, which becomes a bigger refactor.

Reply:
oh, is it inhereted?  i thought it was just a private helper.  i'm fine either way...just prefer to save "operation" for the customer operation framework and use "interaction" when dealing with the FHIR REST API.